src/librustc/lint/builtin.rs

```diff
@@ -2150,6 +2150,42 @@ impl LintPass for HardwiredLints {
     }
 }
 
+declare_lint! {
+    UNEXPORTED_NO_MANGLE,
+    Warn,
+    "functions marked #[no_mangle] should be exported"
+}
+
+#[derive(Copy)]
+pub struct UnexportedNoMangle;
+
+impl LintPass for UnexportedNoMangle {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(UNEXPORTED_NO_MANGLE)
+    }
+
+
+    fn check_item(&mut self, cx: &Context, it: &ast::Item) {
+        #[inline]
+        fn warn(cx: &Context, it: &ast::Item) {
+            cx.span_lint(UNEXPORTED_NO_MANGLE,
+                         it.span,
+                         format!("function {} is marked #[no_mangle], but not exported",
+                                 it.ident).as_slice());
+        }
```

Comment:
I did this because of rightward drift, attempting to avoid upsetting tidy. I'll have a poke.